src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlClientDiagnosticListener.cs

```diff
@@ -56,6 +69,7 @@ public override void OnEventWritten(string name, object? payload)
             case SqlDataBeforeExecuteCommand:
             case SqlMicrosoftBeforeExecuteCommand:
                 {
+                    this.beginTimestamp.Value = Stopwatch.GetTimestamp();
```

Comment:
This should move after `StartActivity` below and should only be necessary when there are no tracing handles.